crates/engine/tree/src/tree/mod.rs

```diff
@@ -2887,17 +2887,23 @@ where
                         state_provider.state_root_with_updates(hashed_state.clone())?;
 
                     if regular_root == sealed_block.header().state_root() {
-                        let provider_ro = state_root_task_config.consistent_view.provider_ro()?;
-                        let in_memory_trie_cursor = InMemoryTrieCursorFactory::new(
-                            DatabaseTrieCursorFactory::new(provider_ro.tx_ref()),
-                            &state_root_task_config.nodes_sorted,
-                        );
-                        compare_trie_updates(
-                            in_memory_trie_cursor,
-                            task_trie_updates.clone(),
-                            regular_updates,
-                        )
-                        .map_err(ProviderError::from)?;
+                        if self.config.always_compare_trie_updates() {
```

Comment:
i think we can safely remove this clause here, so that we always compare trie updates if either state root task root doesn't match the header state root, or the `always_compare_trie_updates` flag is on
```suggestion
```

Reply:
makes sense, done here https://github.com/paradigmxyz/reth/pull/14387/commits/919d640f6cb0eb1bcc83a85d6d4bea46d9532060 PTAL